lib/std/debug.zig

```diff
@@ -167,29 +158,44 @@ pub fn relocateContext(context: *ThreadContext) void {
     };
 }
 
-pub const have_getcontext = @hasDecl(os.system, "getcontext") and
-    (builtin.os.tag != .linux or switch (builtin.cpu.arch) {
-    .x86,
-    .x86_64,
-    => true,
-    else => builtin.link_libc and !builtin.target.isMusl(),
-});
+/// Feature availability.
+///     .{
+///         .enabled // if no frame-pointer tracing is available
+///         .getcontext // if enabled, the context function to use
+///     }
+pub const nofp_tracing = nofpt: {
+    if (builtin.link_libc) {
+        if (builtin.target.isMusl()) {
+            // musl does not implement getcontext
+            if ((builtin.os.tag == .linux) and @hasDecl(os.linux.arch_bits, "getcontext")) {
+                // choose syscall implementation
+                break :nofpt .{ .enabled = true, .getcontext = os.linux.arch_bits.getcontext };
+            } else {
+                break :nofpt .{ .enabled = false };
+            }
+        } else {
+            // choose std.c implementation
+            break :nofpt .{ .enabled = true, .getcontext = std.c.getcontext };
+        }
+    } else if ((builtin.os.tag == .linux) and @hasDecl(os.linux.arch_bits, "getcontext")) {
+        // choose syscall implementation
+        break :nofpt .{ .enabled = true, .getcontext = os.linux.arch_bits.getcontext };
+    } else {
+        break :nofpt .{ .enabled = false };
+    }
+};
```

Comment:
This function should not exist.

Note that `@import("builtin").omit_frame_pointer` exists.